src/sentry/migrations/0397_add_unique_constraint_indexer.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 2.2.28 on 2023-03-16 22:02
+
+from django.db import migrations, models
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = True
+
+    dependencies = [
+        ("sentry", "0396_add_usecase_to_indexer"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[],
+            state_operations=[
+                migrations.AddConstraint(
+                    model_name="perfstringindexer",
+                    constraint=models.UniqueConstraint(
+                        fields=("string", "organization_id", "use_case_id"),
+                        name="perf_unique_org_string_usecase",
+                    ),
+                ),
+                migrations.RemoveConstraint(
+                    model_name="perfstringindexer",
+                    name="perf_unique_org_string",
+                ),
+            ],
+        )
```

Comment:
You'll need to move these operations outside of `SeparateDatabaseAndState`. Having them in `state_operations` causes them to only update the app state, and not change the database.

Reply:
Thanks for catching that

Reply:
the generated SQL looks good, though it doesn't have the USING btree clause

Reply:
I think this is fine. I think the validation pattern @markstory mentioned above only really applies to other types of constraints, like `NOT NULL` etc. Adding the unique index means that inherently there are no duplicates on the table, so I think that all that the constraint needs to do is verify that the unique is in place.